lib/linter.js

```diff
@@ -890,21 +898,25 @@ class Linter extends EventEmitter {
 
             const ecmaFeatures = this.currentConfig.parserOptions.ecmaFeatures || {};
             const ecmaVersion = this.currentConfig.parserOptions.ecmaVersion || 5;
+            const sourceType = this.currentConfig.parserOptions.sourceType || "script";
```

Comment:
Hmm, that seems like a bug to me.

Anyway, it can be addressed in a different issue.